client/src/pages/Map/Map.js

```diff
@@ -0,0 +1,189 @@
+import React, { useState, useEffect, useRef, useCallback } from "react";
+import ReactMapGL, { Marker, Popup } from "react-map-gl";
+import Geocoder from "react-map-gl-geocoder";
+import "react-map-gl-geocoder/dist/mapbox-gl-geocoder.css";
+import "mapbox-gl/dist/mapbox-gl.css";
+import { listSpots } from "../../api/spots";
+import Header from "../../components/Header";
+import Footer from "../../components/Footer";
+import mapMarkerYellow from "../../assets/icons/map-marker-yellow.svg";
+import mapMarkerRed from "../../assets/icons/map-marker-red.svg";
+import SpotEntryForm from "../../components/SpotEntryForm/SpotEntryForm";
+import MapMain from "../../components/MapMain/MapMain";
+import StyledMapContainer from "../../components/StyledMapContainer";
+import SpotEntryFormContainer from "../../components/SpotEntryFormContainer";
+import SpotPopupContainer from "../../components/SpotPopupContainer";
+
+const Map = () => {
+  const [spots, setSpots] = useState([]);
+  const [showPopup, setShowPopup] = useState({});
+  const [addSpot, setAddSpot] = useState(null);
+  const [viewport, setViewport] = useState({
+    height: "100%",
+    width: "100%",
+    latitude: 51.9565,
+    longitude: 10.2755,
+    zoom: 4.5,
+  });
+
+  // This code is used for the Geocoder
+
+  const mapRef = useRef();
+  const handleViewportChange = useCallback(
+    (newViewport) => setViewport(newViewport),
+    []
+  );
+
+  const handleGeocoderViewportChange = useCallback((newViewport) => {
+    const geocoderDefaultOverrides = { transitionDuration: 1000 };
+
+    return handleViewportChange({
+      ...newViewport,
+      ...geocoderDefaultOverrides,
+    });
+  }, []);
+
+  useEffect(() => {
+    (async () => {
+      const spots = await listSpots();
+      setSpots(spots);
+    })();
+  }, []);
+
+  const showAddMarkerPopup = (event) => {
+    const [longitude, latitude] = event.lngLat;
+    setAddSpot({
+      latitude,
+      longitude,
+    });
+  };
+
+  return (
+    <StyledMapContainer>
+      <Header />
+      <MapMain>
+        <ReactMapGL
+          ref={mapRef}
+          {...viewport}
+          mapboxApiAccessToken={
+            "pk.eyJ1IjoibWFudXNhbmNoZXoyIiwiYSI6ImNrZDAyMGF6ajBwOGEzMW91YWhpaXBrd3IifQ.DxIOHXiZw3-rvAn5yC8QYw"
+          }
+          //   mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_TOKEN}
+          mapStyle="mapbox://styles/manusanchez2/ckf3tjjnc10a619s7rnb6iqfa"
```

Comment:
This access token shouldn't be in git. 
If you need client side secrets, see https://create-react-app.dev/docs/adding-custom-environment-variables/.

The comment looks good. Maybe you had an issue?

Reply:
Exactly, at the moment it did not recognize the token, that's why I had to do it this way. I will double check on it!

Reply:
Now it works!!!!